examples/java/src/main/java/org/apache/beam/examples/complete/TfIdf.java

```diff
@@ -179,9 +173,11 @@ public ReadDocuments(Iterable<URI> uris) {
           uriString = uri.toString();
         }
 
-        PCollection<KV<URI, String>> oneUriToLines = pipeline
-            .apply("TextIO.Read(" + uriString + ")", TextIO.read().from(uriString))
-            .apply("WithKeys(" + uriString + ")", WithKeys.<URI, String>of(uri));
+        PCollection<KV<URI, String>> oneUriToLines =
+            pipeline
+                .apply("TextIO.Read(" + uriString + ")", TextIO.read().from(uriString))
+                .apply("WithKeys(" + uriString + ")", WithKeys.<URI, String>of(uri))
+                .setCoder(KvCoder.of(StringDelegateCoder.of(URI.class), StringUtf8Coder.of()));
```

Comment:
We don't have any way around using this (now deprecated) method for cases like this though, right? (Until WithKeys is augmented to take a Coder...I wonder if there's a way to do that generically without having to modify all (existing and future) PTransforms individually. (Maybe it could be an optional argument in apply?)